src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -243,14 +243,24 @@ static SourceGeneratorExecutionVersionMap FilterToProjectCone(SourceGeneratorExe
                 Dictionary<ProjectId, ProjectStateChecksums> newProjectIdToStateChecksums,
                 CancellationToken cancellationToken)
             {
-                // Note: it's common to see a whole lot of project-infos change.  So attempt to collect that in one go
-                // if we can.
-                using var _ = PooledHashSet<Checksum>.GetInstance(out var projectInfoChecksums);
-                foreach (var (projectId, newProjectChecksums) in newProjectIdToStateChecksums)
-                    projectInfoChecksums.Add(newProjectChecksums.Info);
+                // Note: it's common to need to collect a large set of project-attributes and compilation options.  So
+                // attempt to collect all of those in a single call for each kind instead of a call for each instance
+                // needed.
```

Comment:
This code only really needs to execute on changed projects, rignt? If so, could we move this code down there and have it only work on checksums from those projects?

Reply:
changed *and* added projects.  which is precisely what is in newProjectIdToStateChecksums.